### ISSUE TITLE
@snippetdoc not support multi lines

### ISSUE THREAD

**Issue by zhangzq** [doxygen/doxygen#10991]
**Describe the bug**
If I create snippet with multi lines, then It output weird characters and formats.

example code:

```python
"""[test1]
first line

second line
[test1]"""

def foo():
    """! function foo

    @snippet{doc} this test1
    """
    pass
```

**Screenshots**

![image](https://github.com/doxygen/doxygen/assets/3370086/d4d17529-6ba1-4248-a329-bcd89bbec468)

**To Reproduce**

The above piece of codes with default doxyfile.

**Expected behavior**

There should be two lines, first line and second line, without `autotoc_md0`, without bold.

**Version**

1.12.0 (4a9be1303830cb78f8def8b60888cd6e23afd34d*)

**Comment by zhangzq** [comment]
If I write code as following, then it will be OK. It seems that not support style of `"""! comments """` 

```python
"""[test1]
first line

second line
[test1]"""

## function foo
#
# @snippet{doc} this test1
def foo():
    pass
```

**Comment by albert-github** [comment]
With the doxygen 1.11.0 version it all looks OK but with the current master (I used 1.12.0 (93ad49251c5f273a7a276e22ef2f8ef48afdd86c) ) it is as reported and I also get the warning
```
aa.py:7: warning: invalid argument for command '\iline'
```

Looks like it is a problem in the commentcnv.
When having:
```
## \file

"""[test1]
first line

second line
[test1]"""

## function foo_hash
#
# @snippet{doc} this test1
def foo_hash():
    pass

def foo_exclamation():
    """! function foo

    @snippet{doc} this test1
    """
    pass
```
the result is
```
output=[
## \file

"""[test1]
first line

second line
[test1]"""

## function foo_hash
#
#  \ifile ".../aa.py" \iline 4 \ilinebr   first line
#
# second line
#  \ifile ".../aa.py" \iline 11 \ilinebr
def foo_hash():
    pass

def foo_exclamation():
    """! function foo

     \ifile ".../aa.py" \iline 4 \ilinebr     first line
#
#   second line
#    \ifile ".../aa.py" \iline 25 \ilinebr
    """
    pass
]
```
note the `#` signs with the function `foo_exclamation` these should in case of `"""!` not be present

A bisect gave:
```
38ee114a4563fd91ea547c262d120cf361b83039 is the first bad commit
commit 38ee114a4563fd91ea547c262d120cf361b83039

Date:   Sun Jun 23 14:44:54 2024 +0200

    issue #10935 \snippet{doc} tag in Doxygen v1.11 adds incorrect paragraph with a break before snippet text

 src/commentcnv.l     | 165 ++++++++++++++++++++++++++-------------------------
 src/fortranscanner.l |   2 +-
 src/markdown.cpp     |  53 ++++++++++++++++-
 src/pyscanner.l      |   2 +-
 src/trace.h          |   2 +-
 5 files changed, 138 insertions(+), 86 deletions(-)

```

**Comment by doxygen** [comment]
@zhangzq Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by zhangzq** [comment]
> @zhangzq Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

it works. Thank you for the so fast resolve !

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
